libs/media-player/src/lib/audio-annotator.tsx

```diff
@@ -0,0 +1,274 @@
+
+    const [outPointSeconds, setOutPointSeconds] = useState<number | null>(null);
+
+    const [errorMessage, setErrorMessage] = useState<string>('');
+
+    const onTimeUpdate = () => {
+        const currentTime = audioRef.current!.currentTime;
+
+        if (currentTime > 0) {
+            setIsPlayHeadMoved(true);
+        }
+    };
+
+    const markInPoint = () => {
+        setErrorMessage('');
+
+        console.log({ inpoint: audioRef.current!.currentTime });
+
+        setInPointSeconds(audioRef.current!.currentTime);
+
+        toggleTimeRangeSelectionBars('inPointSelected');
+    };
+
+    const markOutPoint = () => {
+        setErrorMessage('');
+
+        console.log({ outpoint: audioRef.current!.currentTime });
+
+        const timeRangeSelection = {
+            inPointSeconds: inPointSeconds as number,
+            outPointSeconds: audioRef.current!.currentTime as number,
+        };
+
+        const duration = audioRef.current!.duration;
+
+        if (!isValidTimeRangeSelection(timeRangeSelection, duration)) {
+            if (timeRangeSelection.inPointSeconds >= duration) {
+                setInPointSeconds(null);
+
+                setOutPointSeconds(null);
+
+                setUxState({ timeRangeSelectionBar: null });
+
+                onTimeRangeSelected(null);
+
+                setErrorMessage(
+                    'Not a valid time range.  Marker points cannot be set after play ends'
+                );
+            } else {
+                setErrorMessage('Not a valid time range.  Out point must come after in point');
+            }
+
+            return;
+        }
+
+        setOutPointSeconds(audioRef.current!.currentTime);
+
+        toggleTimeRangeSelectionBars('fullSelection');
+
+        onTimeRangeSelected(timeRangeSelection);
+    };
+
+    const clearMarkers = () => {
+        setInPointSeconds(null);
+
+        setOutPointSeconds(null);
+
+        setUxState({ timeRangeSelectionBar: null });
+
+        onTimeRangeSelected(null);
+
+        setErrorMessage('');
+    };
+
+    useKeyDown(() => {
+        markInPoint();
+    }, [KeyboardKey.inpoint]);
+
+    useKeyDown(() => {
+        markOutPoint();
+    }, [KeyboardKey.outpoint]);
+
+    useKeyDown(() => {
+        clearMarkers();
+    }, [KeyboardKey.clear]);
+
+    const toggleTimeRangeSelectionBars = (timeRangeSelectionBar: TimeRangeSelectionBar) => {
+        console.log(`timeRangeSelectionBar: ${timeRangeSelectionBar} uxstate: `, uxState);
+
+        const isVisible = isBarVisible(timeRangeSelectionBar, uxState);
+
+        if (isVisible) {
+            setUxState({ timeRangeSelectionBar: null });
+        } else {
+            setUxState({ timeRangeSelectionBar: timeRangeSelectionBar });
+        }
+    };
+
+    return (
+        <Stack>
+            <StyledAudioPlayer ref={audioRef} onTimeUpdate={onTimeUpdate} controls>
+                {isAudioMIMEType(mimeType) ? (
+                    <source key={mimeType} src={audioUrl} type={mimeType} />
```

Comment:
It feels like we could reuse this part of the basic audio player. There must be some way to share this, but let's wait until we write it a third time.

Reply:
Agreed, it shouldn't be hard.  Also note that in Chrome, if you open a local video file directly in the browser, by default it uses the `<video>` element as the player.